### PR TITLE
docs: Add replace dockerfile with Go guide

### LINKED ISSUE #3426: Replace your Dockerfile with go Guide + Demo Video
A guide that shows how to replace a Dockerfile with Go 

**To-do:** 
- [ ] Guide 
- [ ] Demo Video (don't start recording until guide content is approved)

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit adds a guide that explains how to replace a Dockerfile with a Dagger pipeline written in Go. It uses
the Dockerfile for the memcached open-source project.

Closes #3426 

Signed-off-by: Vikram Vaswani <vikram@dagger.io>

### CODE REVIEW

File: docs/current/sdk/go/guides/205271-replace-dockerfile.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+- You have a Go development environment with Go 1.15 or later. If not, [download and install Go](https://go.dev/doc/install).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+- You have a Go module with the Dagger Go SDK installed. If not, [install the Dagger Go SDK](../371491-install.md).
+
+## Step 1: Understand the source Dockerfile
+
+To illustrate the process, this guide replicates the build process for the popular open source [Memcached caching system](https://www.memcached.org/) using Dagger. It uses the Dockerfile for the [official Docker Hub Memcached image](https://github.com/docker-library/memcached).
+
+Begin by reviewing the [Dockerfile](https://github.com/docker-library/memcached/blob/1e3f84629bb2ab9975235401c716c1e00563fa82/alpine/Dockerfile) to understand how it works. This Dockerfile is current at the time of writing and is available under the BSD 3-Clause License.
```
Comment (vikram-dagger) [inline docs/current/sdk/go/guides/205271-replace-dockerfile.md:31]: Copyright/legal question here. Is it OK to copy the source Dockerfile into our tutorial? I've not done it right now out of caution, instead linking to the source commit ID (which is fine except that the user needs an extra click to see it).

Comment (vikram-dagger) [reply docs/current/sdk/go/guides/205271-replace-dockerfile.md]: Update: I think it's better to link to it, so I'll leave this as is unless there are strong feelings that it should change (eg. to make the tutorial easier to follow)

**Review by kpenfound — APPROVED** [review]
Content looks good! How do you feel about the 3 functions in the snippet? Do you feel that it makes the code easier to follow? The goal was to make the main function's logical steps easy to read with the details pulled away a bit

**Comment by vikram-dagger** [comment]
> Content looks good! How do you feel about the 3 functions in the snippet? Do you feel that it makes the code easier to follow? The goal was to make the main function's logical steps easy to read with the details pulled away a bit

The functions do make it clearer. I personally found the division of code between main() and other functions a bit inconsistent ie. I wondered why we didn't also have create setEnvVars() and setEntrypoint() custom functions (or include those steps in the existing functions). Maybe that could be an improvement for future?

**Pull request merged**